src/CookieActionComponent.cpp

```diff
@@ -170,37 +170,62 @@ void CookieActionComponent::checkAndPerformAction(double deltaTime, double total
             gameObj2->changeMaterial(materialManager.getMaterial("Brass"));
             gameObj3->changeMaterial(materialManager.getMaterial("Brass"));
             gameObj4->changeMaterial(materialManager.getMaterial("Brass"));
+            gameObj->setScale(smallScale);
+            gameObj1->setScale(smallScale);
+            gameObj2->setScale(smallScale);
+            gameObj3->setScale(smallScale);
+            gameObj4->setScale(smallScale);
         }
     }
 
 
     glm::vec3 aimTarget = holder_->getPosition() + throwDirection * 0.5f;
     gameObj->setPosition(aimTarget);
+    setAimRotation(gameObj, gameManager );
     if (chargeTime > 0.01 && chargeTime <= 0.3) {
         gameObj->changeMaterial(materialManager.getMaterial("Emerald"));
+        gameObj->setScale(largeScale);
     }
 
     glm::vec3 aimTarget1 = holder_->getPosition() + throwDirection * 1.0f;
     gameObj1->setPosition(aimTarget1);
+    setAimRotation(gameObj1, gameManager);
     if (chargeTime > 0.3 && chargeTime <= 0.6) {
         gameObj1->changeMaterial(materialManager.getMaterial("Emerald"));
+        gameObj1->setScale(largeScale);
     }
 
     glm::vec3 aimTarget2 = holder_->getPosition() + throwDirection * 1.5f;
     gameObj2->setPosition(aimTarget2);
+    setAimRotation(gameObj2, gameManager);
     if (chargeTime > 0.6 && chargeTime <= 0.9) {
         gameObj2->changeMaterial(materialManager.getMaterial("Emerald"));
+        gameObj2->setScale(largeScale);
     }
 
     glm::vec3 aimTarget3 = holder_->getPosition() + throwDirection * 2.0f;
     gameObj3->setPosition(aimTarget3);
+    setAimRotation(gameObj3, gameManager);
     if (chargeTime > 0.9 && chargeTime <= 1.2) {
         gameObj3->changeMaterial(materialManager.getMaterial("Emerald"));
+        gameObj3->setScale(largeScale);
+
     }
 
     glm::vec3 aimTarget4 = holder_->getPosition() + throwDirection * 2.5f;
     gameObj4->setPosition(aimTarget4);
+    setAimRotation(gameObj4, gameManager);
     if (chargeTime > 1.2) {
         gameObj4->changeMaterial(materialManager.getMaterial("Emerald"));
+        gameObj4->setScale(largeScale);
     }
+
+}
+
+void CookieActionComponent::setAimRotation(std::shared_ptr <GameObject> obj, GameManager& gameManager) {
+    std::shared_ptr<GameObject> player = gameManager.getPlayer();
+    float playerRot = player->getYAxisRotation() - player->getOrientAngle() + yRot;
+
+    obj->setYAxisRotation(playerRot);
+    obj->addRotation(xRot, glm::vec3(0.0, 0.0, -1.0));
```

Comment:
tbh, I have no idea why I have to rotate around the negative z axis, but it works.